### PR TITLE
Entire-Code

### PR DESCRIPTION
**Pull request by cheatiiit**
@alma-castant @acoconut : Let me know if any changes are required

### CODE REVIEW

File: Android/PowerUpApp/hs_err_pid18659.log [hunk 1]
```diff
@@ -0,0 +1,521 @@
+#
```
Comment (alma-castant) [inline Android/PowerUpApp/hs_err_pid18659.log]: Do not commit your logs to the repo. Amend the commit removing this file using `git commit --amend`

**Comment by cheatiiit** [comment]
Hi @alma-castant  
Removed the log file
Made a single drawable folder
Changed the layout of PowerUpString
Renamed Activities
Removed unused files
I have intentionally kept the dimension separate so that UI widgets are loosely coupled and changes in size of one in future do not effect the other.

File: Android/PowerUpApp/res/layout/avatar_room.xml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    android:orientation="vertical"
+    android:background="@drawable/background" >
+
+    <ImageView
+        android:id="@+id/eyes"
+        android:layout_width="@dimen/avatar_room_width"
+        android:layout_height="@dimen/avatar_room_height"
+        android:layout_alignParentLeft="true"
+        android:layout_alignParentTop="true"
+        android:layout_marginLeft="33dp"
```
Comment (alma-castant) [inline Android/PowerUpApp/res/layout/avatar_room.xml]: Should be a `@dimen` reference

File: Android/PowerUpApp/res/layout/avatar_room.xml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        android:layout_marginTop="57dp"
```
Comment (alma-castant) [inline Android/PowerUpApp/res/layout/avatar_room.xml]: same

File: Android/PowerUpApp/res/layout/avatar_room.xml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        android:src="@drawable/eye1" />
+
+    <ImageView
+        android:id="@+id/hair"
+        android:layout_width="@dimen/avatar_room_width"
+        android:layout_height="@dimen/avatar_room_height"
+        android:layout_alignTop="@id/eyes"
+        android:layout_marginLeft="35dp"
```
Comment (alma-castant) [inline Android/PowerUpApp/res/layout/avatar_room.xml]: same

File: Android/PowerUpApp/res/layout/avatar_room.xml [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        android:layout_toLeftOf="@id/hair"
+        android:src="@drawable/cloth1" />
+
+    <ImageButton
+        android:id="@+id/faceRight"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_alignRight="@id/face"
+        android:layout_below="@id/face"
+        android:layout_marginTop="21dp"
```
Comment (alma-castant) [inline Android/PowerUpApp/res/layout/avatar_room.xml]: same

File: Android/PowerUpApp/res/layout/avatar_room.xml [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        android:layout_toLeftOf="@id/faceRight"
+        android:src="@drawable/abc_ic_go_search_api_mtrl_alpha" />
+
+    <Button
+        android:id="@+id/continueButtonAvatar"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_alignLeft="@+id/face"
+        android:layout_below="@id/faceRight"
+        android:layout_marginTop="30dp"
```
Comment (alma-castant) [inline Android/PowerUpApp/res/layout/avatar_room.xml]: same

File: Android/PowerUpApp/res/layout/pgame.xml [hunk 7]
```diff
@@ -4,63 +4,85 @@
...
-        android:id="@+id/button1"
+    <TextView
+        android:id="@+id/scenarioNameEditText"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:layout_alignParentBottom="true"
-        android:layout_alignRight="@+id/mainListView"
-        android:layout_marginBottom="17dp"
-        android:text="Back to Map" />
+        android:layout_alignBottom="@+id/redoButton"
```
Comment (alma-castant) [inline Android/PowerUpApp/res/layout/pgame.xml]: Should be only `@id`

File: Android/PowerUpApp/res/layout/pgame.xml [hunk 8]
```diff
@@ -4,63 +4,85 @@
...
         android:id="@+id/mainListView"
-        android:layout_width="200dp"
-        android:layout_height="190dp"
-        android:layout_alignBottom="@+id/button1"
-        android:layout_centerHorizontal="true"
-        android:layout_marginBottom="26dp"
+        android:layout_width="@dimen/main_list_view_width"
+        android:layout_height="@dimen/main_list_view_height"
+        android:layout_alignParentRight="true"
+        android:layout_below="@+id/askerImageView"
```
Comment (alma-castant) [inline Android/PowerUpApp/res/layout/pgame.xml]: Same

File: Android/PowerUpApp/src/com/example/powerupapp/Avatar.java [hunk 9]
```diff
@@ -1,40 +1,94 @@
 package com.example.powerupapp;
```
Comment (alma-castant) [inline Android/PowerUpApp/src/com/example/powerupapp/Avatar.java]: This file can probably be removed now

File: Android/PowerUpApp/src/com/example/powerupapp/Map.java [hunk 10]
```diff
@@ -21,7 +21,7 @@ public void onCreate(Bundle savedInstanceState) {
 		getmDbHandler().open();
```
Comment (alma-castant) [inline Android/PowerUpApp/src/com/example/powerupapp/Map.java]: You probably want to remove this class now

File: Android/PowerUpApp/src/com/example/powerupapp/datamodel/Scenario.java [hunk 11]
```diff
@@ -10,6 +10,7 @@
 	private Integer firstQId;
```
Comment (alma-castant) [inline Android/PowerUpApp/src/com/example/powerupapp/datamodel/Scenario.java]: You could probably just turn all elements and getters into `int`

File: Android/PowerUpApp/src/com/example/powerupapp/db/AbstractDbAdapter.java [hunk 12]
```diff
@@ -17,11 +17,11 @@
 	// Database Version
 	private static final int DATABASE_VERSION = 1;
 	// Database Name
-	private static final String DATABASE_NAME = "DatabaseNamePowerUpApp";
+	private static final String DATABASE_NAME = "DBNamePowerUpAppGSoCProjectchetnaGupta";
```
Comment (alma-castant) [inline Android/PowerUpApp/src/com/example/powerupapp/db/AbstractDbAdapter.java]: The database could have a simple name like `PowerUpDB`

File: Android/PowerUpApp/res/layout/pgame.xml (hunk 7 above)
Comment (cheatiiit) [inline Android/PowerUpApp/res/layout/pgame.xml]: RedoButton has not been declared before this point. Need this +

File: Android/PowerUpApp/res/layout/pgame.xml (hunk 8 above)
Comment (cheatiiit) [inline Android/PowerUpApp/res/layout/pgame.xml]: Need the + here

**Comment by cheatiiit** [comment]
Updated the PR

File: Android/PowerUpApp/res/layout/gamemap.xml [hunk 13]
```diff
@@ -2,57 +2,58 @@
...
 
     <Button
-        android:id="@+id/button1"
+        android:id="@+id/ContinueButtonMapGameBegins"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
         android:layout_alignParentBottom="true"
         android:layout_alignParentRight="true"
-        android:layout_marginRight="52dp"
+        android:layout_marginRight="@dimen/conitnue_button_map_margin_right"
```
Comment (alma-castant) [inline Android/PowerUpApp/res/layout/gamemap.xml]: there's a typo in "continue"

File: Android/PowerUpApp/res/layout/pgame.xml (hunk 7 above)
Comment (alma-castant) [inline Android/PowerUpApp/res/layout/pgame.xml]: Ok

File: Android/PowerUpApp/res/layout/pgame.xml (hunk 8 above)
Comment (alma-castant) [inline Android/PowerUpApp/res/layout/pgame.xml]: Ok

**Comment by cheatiiit** [comment]
Changed color/white to android:color.white. @alma-castant :)

**Comment by alma-castant** [comment]
@rosariorobinson +1

**Pull request merged**